cli/src/debugger.rs

```diff
@@ -111,18 +128,21 @@ impl DebugCli {
             help_text: "Read 32bit value from memory",
 
             function: |cli_data, args| {
-                let address_str = args.get(0).ok_or(CliError::MissingArgument)?;
+                let address = get_int_argument(args, 0)?;
 
-                let address = parse_u32_hex(address_str);
-
-                let num_words = args
-                    .get(1)
-                    .map(|c| c.parse::<usize>().expect("Couldn't parse number of words"))
-                    .unwrap_or(1);
+                let num_words = if args.len() > 1 {
+                    get_int_argument(args, 1)?
+                } else {
+                    1
+                };
 
                 let mut buff = vec![0u32; num_words];
 
-                cli_data.core.read_32(address, &mut buff)?;
+                if num_words > 1 {
+                    cli_data.core.read_32(address, &mut buff)?;
+                } else {
+                    buff[0] = cli_data.core.read_word_32(address)?;
+                }
```

Comment:
I am just curious why you did this distinction; I take no particular issue with it other than it seems more cluttered.

Reply:
I wanted to see if there is a difference between `read_word_32` and `read_32`, while debugging.